### PR TITLE
prometheus-operator: add EndpointSlices for operator to support K8s 1.22+

### PR DESCRIPTION
**Pull request by prudnitskiy**
#### What this PR does / why we need it

This MR adds EndpointSlices permissions for Kube-prometheus-operator to support K8s version 1.22+

#### Which issue this PR fixes

There's no open MR for this repo, but there's an ongoing discussion in Prometheus operator [repository](https://github.com/prometheus-operator/prometheus-operator/issues/3862)

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
Why should we add it now if it's not even implemented upstream?

**Comment by prudnitskiy** [comment]
As far as I see it from my own experience - it is already implemented somehow (?). Without this change prometheus-operator constantly restarts without any log messages.

**Review by Xtigyro — CHANGES_REQUESTED** [review]
Hey @prudnitskiy - thanks for the contribution!

Could you please bump the minor version (instead of the bugfix version)?

File: charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml [hunk 1]
```diff
@@ -78,4 +78,12 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
```
Comment (Xtigyro) [inline charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml:81]: Can we put also an `if` clause around this and have this deployed only when the K8s version is `>=1.22`?

**Review by Xtigyro — CHANGES_REQUESTED** [review]

**Comment by TsengSR** [comment]
> As far as I see it from my own experience - it is already implemented somehow (?). Without this change prometheus-operator constantly restarts without any log messages.

I'm seeing the same issue on the operator after upgrading to 45.0 helm chart. Running against k8s v1.21.7 (can't upgrade right now for several reasons). I tried adding the permissions to the clusterrole myself, but the service is still crashing

File: charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml (hunk 1 above)
Comment (TsengSR) [reply charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml:81]: According to the [k8s docs](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) endpoint slices are a 1.21 feature

Comment (monotek) [reply charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml:81]: But the chart claims to be k8s 1.16 compatbile: https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#prerequisites

Comment (prudnitskiy) [reply charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml:83]: Case added, please check

**Review by monotek — APPROVED** [review]

**Comment by istheway** [comment]
For reference, this is problematic for k8s 1.26.2 that I just attempted deploying the latest operator with the kube-prometheus-stack helm chart.  The kube-prometheus-stack-operator CrashLoopBackoff on the following logs:

`level=warn ts=2023-04-18T23:37:00.703117184Z caller=operator.go:332 component=prometheusoperator msg="failed to check if the API supports the endpointslice resources" err="converting (v1.APIGroup) to (v1.APIResourceList): unknown conversion"`

What's interesting is the bind address tcp 10250 is not in use at the moment on any service.

**Pull request merged**